cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -83,6 +91,92 @@ public void download(String url, Path target) {
     }
   }
 
+  /**
+   * Downloads a file while showing a progress bar
+   *
+   * @param url the url to download
+   * @param target Path of the target directory
+   * @param response {@link HttpResponse} to use
+   */
+  private void downloadFileWithProgressBar(String url, Path target, HttpResponse<InputStream> response) throws IOException {
+
+    String contentLength = response.headers().firstValue("content-length").orElse("undefined");
+    if (contentLength.equals("undefined")) {
+      throw new IllegalStateException("Content-Length was not provided by download source: " + url);
+    }
+
+    long size = Long.parseLong(contentLength);
```

Comment:
Please use `firstValueAsLong` directly instead of manual parsing. Also consider that the server could send an invalid content header like `31MB`. I would prefer to be as robust as possible and rather catch such exceptions falling back to some default behavior (e.g. assume a download size of 10MB and having a progress bar that does not work smooth or even skip the process bar entirely).

Reply:
I've added a fallback to 10MB now.